### ISSUE TITLE
Assigning JSXGraph Point coordinates to Typescript Variables. 

### ISSUE THREAD

**Issue by msheakc** [stemcstudio/stemcstudio#69]
While I am working on the Geometric Algebra videos, which are quite good. I am learning a lot. 

I am also experimenting with other uses of StemCstudio and JSXGraph. I have been a user of Geogebra for a long time but, when trying to make sophisticated instructional materials using Javascript I am frustrated by numerous limitations and poor design decisions in the programs JavaScript scripting support.  I think StemCstudio might be the solution I need. 

As a first step I am trying to assign the coordinates of a JSXGraph Point to a typescript variable to use in calculation.  As you can see in [https://www.stemcstudio.com/gists/7283e509c9b5ed0aea03ff65b4ec2c80](url)  it works fine, but keep getting **"Property X does not exist on type GeometryElement**" and similar errors.  I searched the web for answers, but did not find any. I think it has to with using TypeScript and not JavaScript with JSXGraph. 

Any help you can give me would be greatly appreciated.

**Comment by geometryzen** [comment]
@msheakc 
I know we talked about this over email, but I'll complete the resolution here. The problem is that the JsxGraph API design makes it rather tricky to get the type analysis experience good for the developer. I've recently provided a book as the User Guide available from the home page. In there I discuss the issue. See "The overloaded Board.create method and pitfalls when using TypeScript". The root of the problem is that the JsxGraph create API uses parameters to infer the return type. Any error in the parameters throws off the type an analysis. A good solution is to write your own wrapper or helper functions for the JsxGraph the have functions like "createPoint(parents: unknown[], attributes?: PointAttributes): Point.
Good Luck
